smithy-model/src/main/java/software/amazon/smithy/model/validation/ValidationUtils.java

```diff
@@ -26,21 +26,89 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 import java.util.StringJoiner;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import software.amazon.smithy.model.shapes.ShapeId;
 import software.amazon.smithy.model.shapes.ToShapeId;
+import software.amazon.smithy.model.validation.validators.EnumShapeValidator;
+import software.amazon.smithy.model.validation.validators.ExclusiveStructureMemberTraitValidator;
+import software.amazon.smithy.model.validation.validators.ResourceCycleValidator;
+import software.amazon.smithy.model.validation.validators.ShapeIdConflictValidator;
+import software.amazon.smithy.model.validation.validators.SingleOperationBindingValidator;
+import software.amazon.smithy.model.validation.validators.SingleResourceBindingValidator;
+import software.amazon.smithy.model.validation.validators.TargetValidator;
+import software.amazon.smithy.model.validation.validators.TraitConflictValidator;
+import software.amazon.smithy.model.validation.validators.TraitTargetValidator;
+import software.amazon.smithy.model.validation.validators.TraitValueValidator;
+import software.amazon.smithy.utils.SetUtils;
 
 /**
  * Utility methods used when validating.
  */
 public final class ValidationUtils {
+
+    static final Set<Class<? extends Validator>> CRITICAL_VALIDATORS = SetUtils.of(
+            // These validators should run before even any other validators. These validators are considered
+            // correctness validators that ensure relationships are valid and to shapes that can be found and that
+            // the model has no cycles.
+            TargetValidator.class,
+            ResourceCycleValidator.class,
+
+            // This validator is considered critical because it ensures an enum shape itself is valid.
+            EnumShapeValidator.class,
+
+            // Ensure that traits are valid, according to the model, before other validators are run.
+            ExclusiveStructureMemberTraitValidator.class,
+            TraitTargetValidator.class,
+            TraitValueValidator.class,
+            TraitConflictValidator.class,
+
+            // Ensure The semantic model adheres to the requirements of shape IDs and service shape uniqueness.
+            ShapeIdConflictValidator.class,
+            SingleOperationBindingValidator.class,
+            SingleResourceBindingValidator.class
+    );
```

Comment:
This is just the set I came up with and I'm happy to add more or remove some if needed.

Reply:
I think this is the right list. Here are a couple others I looked at for inclusion but don't think they should be added.

* `PrivateAccessValidator`
  * Akin to `TargetValidator`, the shape is invalid if it has a relationship to something in a private namespace. Counterpoint: this isn't as bad since `TargetValidator` would catch the thing that breaks downstream validation.
* `ShapeRecursionValidator`
  * Improperly recursive shapes being walked during validation can cause infinite recursions. Counterpoint: this is still the case without invalid recursion, as validators that need to walk the graph have to track valid shape recursion.